File: tps/entrada e saida/tp02/TP02Q01 - Classe Filme em Java/tmp/TP02Q01.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.*;

public class TP02Q01 {

    public static void main(String[] args) {
        MyIO.setCharset("UTF-8");

        String[] entrada = new String[1000];
        int numEntrada = 0;

        do {
            entrada[numEntrada] = MyIO.readLine();
        } while (!(isFim(entrada[numEntrada++])));
        numEntrada--;

        ArrayList<String> fileLines = lerArquivo("tmp/games.csv");
        for (int i = 0; i < numEntrada; i++) {
            for (String line : fileLines) {
                if (entrada[i].equals(line.split(",")[0])) {
                    Game game = new Game();
                    game.ler(line);
                    game.imprimir();
                }
            }
        }
    }

    public static boolean isFim(String str) {
        return ((str.length() == 3) && (str.charAt(0) == 'F') && (str.charAt(1) == 'I')
                && (str.charAt(2) == 'M'));
    }

    public static ArrayList<String> lerArquivo(String fileName) {
        ArrayList<String> linhas = new ArrayList<String>();
        try {
            FileReader arq = new FileReader(fileName);
            BufferedReader br = new BufferedReader(arq);

            for (String line = br.readLine(); line != null; line = br.readLine()) {
                linhas.add(line);
            }

            br.close();
        } catch (IOException ioe) {
        }

        return linhas;
    }
}

class Game {
    private int app_id;
    private String name;
    private Date release_date;
    private String owners;
    private int age;
    private float price;
    private int dlcs;
    private String[] languages;
    private String website;
    private boolean windows;
    private boolean mac;
    private boolean linux;
    private float upvotes;
    private int avg_pt;
    private String developers;
    private String[] genres;

    public Game() {
        this.app_id = -1;
        this.name = "";
        this.release_date = new Date();
        this.owners = "";
        this.age = 0;
        this.price = 0.0f;
        this.dlcs = 0;
        this.languages = new String[0];
        this.website = "";
        this.windows = false;
        this.mac = false;
        this.linux = false;
        this.upvotes = 0.0f;
        this.avg_pt = 0;
        this.developers = "";
        this.genres = new String[0];
    }

    public Game(
            int app_id, String name, Date release_date, String owners,
            int age, float price, int dlcs, String[] languages,
            String website, boolean windows, boolean mac, boolean linux,
            float upvotes, int avg_pt, String developers, String[] genres) {
        this.app_id = app_id;
        this.name = name;
        this.release_date = release_date;
        this.owners = owners;
        this.age = age;
        this.price = price;
        this.dlcs = dlcs;
        this.languages = languages;
        this.website = website;
        this.windows = windows;
        this.mac = mac;
        this.linux = linux;
        this.upvotes = upvotes;
        this.avg_pt = avg_pt;
        this.developers = developers;
        this.genres = genres;
    }

    public Game clone() {
        return new Game(
                this.app_id, this.name, this.release_date, this.owners,
                this.age, this.price, this.dlcs, this.languages.clone(),
                this.website, this.windows, this.mac, this.linux,
                this.upvotes, this.avg_pt, this.developers, this.genres.clone());
    }

    public int getApp_id() {
        return this.app_id;
    }

    public String getName() {
        return this.name;
    }

    public Date getRelease_date() {
        return this.release_date;
    }

    public String getOwners() {
        return this.owners;
    }

    public int getAge() {
        return this.age;
    }

    public float getPrice() {
        return this.price;
    }

    public int getDlcs() {
        return this.dlcs;
    }

    public String[] getLanguages() {
        return this.languages;
    }

    public String getWebsite() {
        return this.website;
    }

    public boolean getWindows() {
        return this.windows;
    }

    public boolean getMac() {
        return this.mac;
    }

    public boolean getLinux() {
        return this.linux;
    }

    public float getUpvotes() {
        return this.upvotes;
    }

    public int getAvg_pt() {
        return this.avg_pt;
    }

    public String getDevelopers() {
        return this.developers;
    }

    public String[] getGenres() {
        return this.genres;
    }

    public void setApp_id(int app_id) {
        this.app_id = app_id;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setRelease_date(Date release_date) {
        this.release_date = release_date;
    }

    public void setOwners(String owners) {
        this.owners = owners;
    }

    public void setAge(int age) {
        this.age = age;
    }

    public void setPrice(float price) {
        this.price = price;
    }

    public void setDlcs(int dlcs) {
        this.dlcs = dlcs;
    }

    public void setLanguages(String[] languages) {
        this.languages = languages;
    }

    public void setWebsite(String website) {
        this.website = website;
    }

    public void setWindows(boolean windows) {
        this.windows = windows;
    }

    public void setMac(boolean mac) {
        this.mac = mac;
    }

    public void setLinux(boolean linux) {
        this.linux = linux;
    }

    public void setUpvotes(float upvotes) {
        this.upvotes = upvotes;
    }

    public void setAvg_pt(int avg_pt) {
        this.avg_pt = avg_pt;
    }

    public void setDevelopers(String developers) {
        this.developers = developers;
    }

    public void setGenres(String[] genres) {
        this.genres = genres;
    }


    public void ler(String rawLine) {
        String[] lineArray = new String[55];
        int charAfterCommaIndex, lineArrayIndex;
        charAfterCommaIndex = lineArrayIndex = 0;

        for (int i = 0; i < rawLine.length(); i++) {
            if (rawLine.charAt(i) == '"') {
                charAfterCommaIndex = ++i;
                do {
                } while (rawLine.charAt(i++) != '"');
                lineArray[lineArrayIndex] = rawLine.substring(charAfterCommaIndex, i - 1);
                charAfterCommaIndex = i + 1;
                lineArrayIndex++;
            } else if (rawLine.charAt(i) == ',') {
                lineArray[lineArrayIndex] = rawLine.substring(charAfterCommaIndex, i);
                charAfterCommaIndex = i + 1;
                lineArrayIndex++;
            } else if (i == rawLine.length() - 1) {
                lineArray[lineArrayIndex] = rawLine.substring(charAfterCommaIndex, i + 1);
            }
        }

        this.app_id = Integer.parseInt(lineArray[0]);
        this.name = lineArray[1];
        try {
            this.release_date = new SimpleDateFormat("MMM dd, yyyy", Locale.ENGLISH).parse(lineArray[2]);
        } catch (ParseException e) {
        }
        this.owners = lineArray[3];
        this.age = Integer.parseInt(lineArray[4]);
        this.price = Float.parseFloat(lineArray[5]);
        this.dlcs = Integer.parseInt(lineArray[6]);
        this.languages = this.stringToStringArray(lineArray[7]);
        this.website = lineArray[8];
        this.windows = Boolean.parseBoolean(lineArray[9]);
        this.mac = Boolean.parseBoolean(lineArray[10]);
        this.linux = Boolean.parseBoolean(lineArray[11]);
        this.upvotes = Float.parseFloat(lineArray[12]) / (Float.parseFloat(lineArray[12]) + Float.parseFloat(lineArray[13]));
        this.upvotes = (float) (Math.round(this.upvotes * 100.0) / 100.0);
        this.avg_pt = Integer.parseInt(lineArray[14]);
        this.developers = lineArray[15];
        this.genres = this.stringToStringArray(lineArray[16]);
    }

    private String[] stringToStringArray(String str) {
        // if (str == null) return null;
        if (str.charAt(0) == '[' && str.charAt(str.length() - 1) == ']'){
            str = str.substring(1, str.length() - 1);
            str = str.replace("'", "").replace(", ", ",");
        }
        return str.split(",");
    }

    private String intToTimeString(int intTime) {
        if (intTime == 0)
            return null;
        int hours = intTime / 60;
        int minutes = intTime - hours * 60;
        String stringTime = "";
        if (hours != 0 && minutes != 0)
            stringTime = Integer.toString(hours) + "h " + Integer.toString(minutes) + 'm';
        else if (hours != 0)
            stringTime = Integer.toString(hours) + 'h';
        else if (minutes != 0)
            stringTime += Integer.toString(minutes) + 'm';
        else
            stringTime = null;
        return stringTime;
    }

    public void imprimir() {
        String[] stringArray = this.release_date.toString().split(" ");
        String dateString = stringArray[1] + '/' + stringArray[stringArray.length - 1];
		String price = Float.toString(this.price);
		if (price.charAt(price.length() - 2) == '.' && price.charAt(price.length() - 1) == '0')
			price += '0';
        String website = this.website.length() == 0 ? null : this.website;
        String upvotes = Integer.toString((int) (this.upvotes * 100)) + '%';
        String avg_pt = this.intToTimeString(this.avg_pt);

		MyIO.println(
            this.app_id + " " + this.name + " " + dateString + " " +
			this.owners + " " + this.age + " " + price + " " +
			this.dlcs + " " + Arrays.toString(this.languages) + " " + website + " " +
			this.windows + " " + this.mac + " " + this.linux + " " +
			upvotes + " " + avg_pt + " " + this.developers + " " +
			Arrays.toString(this.genres)
		);
	}
}

File: tps/entrada e saida/tp02/TP02Q02 - Registro em C/TP02Q01.cpp
#include "TP02Q01.h"
using namespace 

using BufferedReader = java::io::BufferedReader;
using FileReader = java::io::FileReader;
using IOException = java::io::IOException;
using ParseException = java::text::ParseException;
using SimpleDateFormat = java::text::SimpleDateFormat;

void TP02Q01::main(vector<wstring> &args)
{
	MyIO::setCharset(L"UTF-8");

	vector<wstring> entrada(1000);
	int numEntrada = 0;

	do
	{
		entrada[numEntrada] = MyIO::readLine();
	} while (!(isFim(entrada[numEntrada++])));
	numEntrada--;

	vector<wstring> fileLines = lerArquivo(L"tmp/games.csv");
	for (int i = 0; i < numEntrada; i++)
	{
		for (auto line : fileLines)
		{
			if (entrada[i] == line.split(L",")[0])
			{
				Game *game = new Game();
				game->ler(line);
				game->imprimir();

				delete game;
			}
		}
	}
}

bool TP02Q01::isFim(const wstring &str)
{
	return ((str.length() == 3) && (str[0] == L'F') && (str[1] == L'I') && (str[2] == L'M'));
}

vector<wstring> TP02Q01::lerArquivo(const wstring &fileName)
{
	vector<wstring> linhas;
	try
	{
		FileReader *arq = new FileReader(fileName);
		BufferedReader *br = new BufferedReader(arq);

		for (wstring line = br->readLine(); line != L""; line = br->readLine())
		{
			linhas.push_back(line);
		}

		br->close();

		delete br;
//JAVA TO C++ CONVERTER TODO TASK: A 'delete arq' statement was not added since arq was passed to a method or constructor. Handle memory management manually.
	}
	catch (const IOException &ioe)
	{
	}

	return linhas;
}

Game::Game()
{
	this->app_id = -1;
	this->name = L"";
	this->release_date = Date();
	this->owners = L"";
	this->age = 0;
	this->price = 0.0f;
	this->dlcs = 0;
	this->languages = vector<wstring>(0);
	this->website = L"";
	this->windows = false;
	this->mac = false;
	this->linux = false;
	this->upvotes = 0.0f;
	this->avg_pt = 0;
	this->developers = L"";
	this->genres = vector<wstring>(0);
}

Game::Game(int app_id, const wstring &name, Date release_date, const wstring &owners, int age, float price, int dlcs, vector<wstring> &languages, const wstring &website, bool windows, bool mac, bool linux, float upvotes, int avg_pt, const wstring &developers, vector<wstring> &genres)
{
	this->app_id = app_id;
	this->name = name;
	this->release_date = release_date;
	this->owners = owners;
	this->age = age;
	this->price = price;
	this->dlcs = dlcs;
	this->languages = languages;
	this->website = website;
	this->windows = windows;
	this->mac = mac;
	this->linux = linux;
	this->upvotes = upvotes;
	this->avg_pt = avg_pt;
	this->developers = developers;
	this->genres = genres;
}

Game *Game::clone()
{
	return new Game(this->app_id, this->name, this->release_date, this->owners, this->age, this->price, this->dlcs, this->languages.clone(), this->website, this->windows, this->mac, this->linux, this->upvotes, this->avg_pt, this->developers, this->genres.clone());
}

int Game::getApp_id()
{
	return this->app_id;
}

wstring Game::getName()
{
	return this->name;
}

Date Game::getRelease_date()
{
	return this->release_date;
}

wstring Game::getOwners()
{
	return this->owners;
}

int Game::getAge()
{
	return this->age;
}

float Game::getPrice()
{
	return this->price;
}

int Game::getDlcs()
{
	return this->dlcs;
}

vector<wstring> Game::getLanguages()
{
	return this->languages;
}

wstring Game::getWebsite()
{
	return this->website;
}

bool Game::getWindows()
{
	return this->windows;
}

bool Game::getMac()
{
	return this->mac;
}

bool Game::getLinux()
{
	return this->linux;
}

float Game::getUpvotes()
{
	return this->upvotes;
}

int Game::getAvg_pt()
{
	return this->avg_pt;
}

wstring Game::getDevelopers()
{
	return this->developers;
}


//====================================================================================================
//End of the allowed output for the Free Edition of Java to C++ Converter.

//To purchase the Premium Edition, visit our website:
//https://www.tangiblesoftwaresolutions.com/order/order-java-to-cplus.html
//====================================================================================================
